### DISCUSSION TITLE
Camera extension could allow for multiple cameras

### DISCUSSION THREAD

**Discussion by gomigg07** [TurboWarp/extensions#399] [Ideas]
When using the camera extension, I want some sprites to stay in place (like a HUD for example). I could simply make a custom block that changes the position and size with the camera in order to appear in place, but it is inconvenient to copy the script for each sprite I want to do that with. This can also be difficult with a text engine, for example. Spacing between letters varies because of the camera and clone size.

An addition to the extension I would like to propose is a new block:
`[append to camera (#)]`

This block could assign different cameras to certain sprites. For example, any sprites assigned to gameplay use camera 1, any sprites assigned to the HUD use camera 2, anything assigned to menus uses camera 3, etc. This way, I would not need to take care of positioning and sizing scripts for specific sprites.
In my opinion I would limit the amount of cameras to 4 or 5.

I understand if coding something like this is currently impossible, as the Scratch engine only uses one canvas (maybe 2 if you count pen as seperate).

**Comment by plasmatopia** [comment]
Yeah this will help a lot Can you do this

**Comment by LilyMakesThings** [comment]
I think there's a way I could do this but it would require a total rewrite of the extension to be good

Maybe one day
